lib/puppet/node.rb

```diff
@@ -23,8 +23,8 @@ class Puppet::Node
 
   def initialize_from_hash(data)
     @name = data['name']
-    @classes = data['classes']
-    @parameters = data['parameters']
+    @classes = data['classes'] || []
+    @parameters = data['parameters'] || {}
     @environment_name = data['environment']
```

Comment:
Actually, the name is already checked by self.from_data_hash - I will instead mark the initialize_from_hash as an @api private method